test/before.test.js

```diff
@@ -0,0 +1,70 @@
+import assert from 'assert'
+import lodashStable from 'lodash'
+import { _ } from './utils.js'
+
+describe('before', () => {
+  const callMethod = (n, times) => {
+    let count = 0
+
+    lodashStable.times(
+      times,
+      _.before(n, () => {
+        count++
+      })
+    )
+    return count
+  }
+
+  describe('if `func` arg is a valid function', () => {
+    describe("if n isn't a numeric value", () => {
+      it('should coerce NaN values to 0', () => {
+        assert.strictEqual(callMethod(NaN, 10), 0)
+      })
+    })
+
+    describe('if n is a numeric value', () => {
+      describe('when n = 0', () => {
+        it('should create a function that does not invoke `func` when called', () => {
+          assert.strictEqual(callMethod(0, 10), 0)
+        })
+
+        it('should create a function that does not invoke `func` immediately', () => {
+          assert.strictEqual(callMethod(0, 0), 0)
+        })
+      })
+
+      describe('when n > 0', () => {
+        it('should create a function that invokes `func` before being called n times', () => {
+          assert.strictEqual(callMethod(5, 4), 4)
+        })
+
+        it('should create a function that does not invoke `func` after being called n - 1 times', () => {
+          assert.strictEqual(callMethod(5, 5), 4)
+        })
```

Comment:
Perhaps the description here could be improved. For instance, should the _before_ and _after_ terms be considered inclusive or exclusive?

This also impacts whether we say `n` or `n - 1`. Personally, I find it confusing at first to have the first one `n` and the second `n - 1`.